packages/ethereum-group-pcd/src/EthereumGroupPCD.ts

```diff
@@ -0,0 +1,314 @@
+if (isNode) {
+  addrMembershipConfig = {
+    circuit: __dirname.concat("/../artifacts/addr_membership.circuit"),
+    witnessGenWasm: __dirname.concat("/../artifacts/addr_membership.wasm"),
+  };
+
+  pubkeyMembershipConfig = {
+    circuit: __dirname.concat("/../artifacts/pubkey_membership.circuit"),
+    witnessGenWasm: __dirname.concat("/../artifacts/pubkey_membership.wasm"),
+  };
+} else {
+  addrMembershipConfig = {
+    circuit:
+      "https://storage.googleapis.com/personae-proving-keys/membership/addr_membership.circuit",
+    witnessGenWasm:
+      "https://storage.googleapis.com/personae-proving-keys/membership/addr_membership.wasm",
+  };
+
+  pubkeyMembershipConfig = {
+    circuit:
+      "https://storage.googleapis.com/personae-proving-keys/membership/pubkey_membership.circuit",
+    witnessGenWasm:
+      "https://storage.googleapis.com/personae-proving-keys/membership/pubkey_membership.wasm",
+  };
+}
+
+export async function prove(
+  args: EthereumGroupPCDArgs
+): Promise<EthereumGroupPCD> {
+  if (args.identity.value === undefined) {
+    throw new Error(`missing argument identity`);
+  }
+
+  if (args.signatureOfIdentityCommitment.value === undefined) {
+    throw new Error(`missing argument signatureOfIdentityCommitment`);
+  }
+
+  if (args.merkleProof.value === undefined) {
+    throw new Error(`missing argument merkleProof`);
+  }
+
+  const deserializedIdentity = await SemaphoreIdentityPCDPackage.deserialize(
+    args.identity.value.pcd
+  );
+  const message = deserializedIdentity.claim.identity.commitment.toString();
+  const msgHash = ethers.utils.hashMessage(message);
+
+  const prover = new MembershipProver(pubkeyMembershipConfig);
+  await prover.initWasm();
+  const { proof, publicInput } = await prover.prove(
+    args.signatureOfIdentityCommitment.value,
+    Buffer.from(msgHash.slice(2), "hex"),
+    JSONBig({ useNativeBigInt: true }).parse(args.merkleProof.value)
+  );
+
+  const publicInputMsgHash = "0x" + publicInput.msgHash.toString("hex");
+
+  if (msgHash !== publicInputMsgHash) {
+    throw new Error(
+      `public input message hash ${publicInputMsgHash} does not match commitment ${message} hash ${msgHash} `
+    );
+  }
+
+  const semaphoreSignature = await SemaphoreSignaturePCDPackage.prove({
+    identity: {
+      argumentType: ArgumentTypeName.PCD,
+      pcdType: SemaphoreIdentityPCDTypeName,
+      value: args.identity.value,
+    },
+    signedMessage: {
+      argumentType: ArgumentTypeName.String,
+      value: Buffer.from(proof).toString("hex"),
+    },
+  });
+
+  return new EthereumGroupPCD(
+    uuid(),
+    {
+      publicInput: publicInput,
+      groupType:
+        args.groupType.value === "address"
+          ? GroupType.ADDRESS
+          : GroupType.PUBLICKEY,
+    },
+    {
+      signatureProof: await SemaphoreSignaturePCDPackage.serialize(
+        semaphoreSignature
+      ),
+      ethereumGroupProof: Buffer.from(proof).toString("hex"),
```

Comment:
This is too large to fit in a URL. The *last* step of adding the proof shows this screen:

![image](https://github.com/djma/zupass/assets/692818/59e1ce1d-b641-448e-8571-5975ce0eb9a2)

```
localhost:3001/popup?proofUrl=http%3A%2F%2Flocalhost%3A3000%2F%23%2Fprove%3Frequest%3D%257B%2522type%2522%253A%2522Get%2522%252C%2522returnUrl%2522%253A%2522http%253A%252F%252Flocalhost%253A3001%252Fpopup%2522%252C%2522args%2522%253A%257B%2522identity%2522%253A%257B%2522argumentType%2522%253A%2522PCD%2522%252C%2522pcdType%2522%253A%2522semaphore-identity-pcd%2522%252C%2522userProvided%2522%253Atrue%257D%252C%2522signedMessage%2522%253A%257B%2522argumentType%2522%253A%2522String%2522%257D%257D%252C%2522pcdType%2522%253A%2522semaphore-signature-pcd%2522%252C%2522options%2522%253A%257B%2522title%2522%253A%2522Zuzalu%2520Auth%2522%252C%2522description%2522%253A%2522eth-group-pcd%2522%252C%2522signIn%2522%253Atrue%257D%257D
```

But the PCD seems to get added successfully anyways:

![image](https://github.com/djma/zupass/assets/692818/52552e11-46b2-418e-a6eb-7bcdd14d17fb)